### PR TITLE
Support images

### LINKED ISSUE #1: Support images in incoming and outgoing messages
For now, we only support text messages. The [Node](https://github.com/wireapp/bot-sdk-node) and [Java](https://github.com/wireapp/lithium) implementations already support it.

### PR DESCRIPTION
**Pull request by wafflespeanut**
fixes #1

### CODE REVIEW

**Comment by wafflespeanut** [comment]
Note that this is on top of #3

File: src/client.rs [hunk 1]
```diff
@@ -14,34 +15,66 @@ use std::collections::HashMap;
...
+const MULTIPART_BOUNDARY: &'static str = "frontier";
+lazy_static! {
+    static ref MULTIPART_MIXED: Mime = {
+        let mime_str = format!("multipart/mixed; boundary={}", MULTIPART_BOUNDARY);
+        mime_str.parse().unwrap()
+    };
+}
+
+header! {
+    (ContentMd5, "Content-MD5") => [String]     // base64-encoded MD5 hash digest
```
Comment (siilime) [inline src/client.rs]: Arrrgghhhh!

File: src/client.rs [hunk 2]
```diff
@@ -208,27 +239,81 @@ impl HttpsClient {
         message.set_message_id(uuid.to_string());
         let mut confirmation = Confirmation::new();
         confirmation.set_message_id(message_id.to_owned());
-        confirmation.set_field_type(ConfirmationType::DELIVERED);
+        confirmation.set_field_type(Confirmation_Type::DELIVERED);
         message.set_confirmation(confirmation);
         self.send_encrypted_message(client, &message, storage, devices)
     }
+
```
Comment (siilime) [inline src/client.rs]: Ugh. This is rather cumbersome. Will think about how this is made more elegant.

File: src/client.rs [hunk 3]
```diff
@@ -287,4 +373,54 @@ impl BotClient {
             error!("Cannot queue user message in event loop: {}", e);
         }).ok();
     }
+
+    /// Send an user image to the associated conversation. Use the exported
+    /// `Image` to open an image (from path, reader, or buffer).
+    pub fn send_image(&self, img: Arc<Image>) {
```
Comment (siilime) [inline src/client.rs]: Again, cumbersome, but necessary. Might be a more elegant option.

**Pull request merged**